src/extension.ts

```diff
@@ -1731,9 +1736,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
     }
 }
 
-async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+async function useNamespaceKubernetes(explorerNode?: explorer.KubernetesObject) {
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], 'get', undefined, (resource) => {
+            if (resource) {
+                const currentNS = kubectlUtils.currentNamespace(kubectl);
+                currentNS.then((ns) => {
+                    //resource will be of format <kind>/<name>, hence get the substring
```

Comment:
It sounds like you're expecting the user, after invoking Set Namespace, to type `ns/foo` rather than just `foo`.  That seems counterintuitive - the user expects to enter the _namespace_ to switch to, and will not expect to have to enter the `ns/` prefix.

Reply:
no i don't expect that way here with currentNamespace function does return just NS alone and when the explorer.id it returned with `<resource>/name`